Archipelago.MultiClient.Net.Analyzers/Archipelago.MultiClient.Net.Analyzers/Analyzers/ItemFlagsComparisonInSwitchCaseDiagnostics.cs

```diff
@@ -35,15 +36,23 @@ public override void Initialize(AnalysisContext context)
 
         private void AnalyzeSwitch(SyntaxNodeAnalysisContext context)
         {
+            // analyze the switch statement for any cases which compare directly to the ItemFlags enum
             SwitchStatementSyntax syntax = (SwitchStatementSyntax)context.Node;
-            TypeInfo switchType = context.SemanticModel.GetTypeInfo(syntax.Expression, context.CancellationToken);
-
-            if (!ArchipelagoTypeUtils.IsTypeItemFlags(switchType.Type, context.Compilation))
+            foreach (SwitchSectionSyntax section in syntax.Sections)
             {
-                return;
+                foreach (CaseSwitchLabelSyntax label in section.Labels.OfType<CaseSwitchLabelSyntax>())
+                {
+                    if (label.Value is MemberAccessExpressionSyntax identifier)
```

Comment:
There are potentially ways to slip through this with badly behaved (non hasflags) pattern matches, I'm not sure how reasonable or common they are though. It might be worth it to flag any pattern matching that isn't the kind we know is good (obviously we cannot offer a fix but at least we can raise the signal)

Reply:
After thinking on this and looking at the current state of the analyzer now that it accommodates switch expressions, I don't want it to get too complicated and we already have some signals that overly aggressive warnings on other analyzers are annoying. So I will leave it out.